extensions/mongodb-client/runtime/src/main/java/io/quarkus/mongodb/runtime/MongodbConfig.java

```diff
@@ -0,0 +1,42 @@
+package io.quarkus.mongodb.runtime;
+
+import java.util.Map;
+
+import io.quarkus.runtime.annotations.ConfigItem;
+import io.quarkus.runtime.annotations.ConfigPhase;
+import io.quarkus.runtime.annotations.ConfigRoot;
+
+@ConfigRoot(name = "mongodb", phase = ConfigPhase.BUILD_AND_RUN_TIME_FIXED)
```

Comment:
I think the phase should be runt time as before as we want all properties related to the connection to be runtime properties

Reply:
If I switch to runtime only, I have issue with the producer bean. Because I need to loop over named config to create a produce function for each one but I can on build time use runtime config. So 2 solutions: 
- create a build time config with only the list of named connection
- switch runtime config to BUILD_AND_RUN_TIME_FIXED config
wdyt ? @loicmathieu

Reply:
@dufoli BUILD_AND_RUN_TIME_FIXED is OK, you can keep it like this

Reply:
I still think this is incorrect.

Master uses:

```
@ConfigRoot(name = "mongodb", phase = ConfigPhase.RUN_TIME)
public class MongoClientConfig {

}
```

but now it seems like we are changing the configuration to runtime fixed? How will the URL be overridable at runtime if this is the case?

Reply:
@loicmathieu @dufoli have you tried this PR and override some mongo options at runtime?

Reply:
This is absolutely critical for this to clearified (and IMHO fixed)

Reply:
1) Agroal use same config with named config build and runtime.
2) I was thinking that we have to favor build time work over runtime work.
in order to speed up launch by moving runtime stuff to deployment phase right ? So move injection on deployment side is a good move no ?
3) Configuration of a connection is in general setup at start and change rarely on environment in company. If people need to change they can rebuild by the way ?

Reply:
So am I missing some runtime configuration in the PR? Because I see that master has `MongoClientConfig` which was runtime configuration which now seems to have become `runtime fixed`, correct?
If I am missing something sorry.

Although I might agree with some of the properties might be good to be build-time, the fact is that we cannot change runtime properties to build-time properties because that would break user applications and we can't do that in a post `1.0` world. 
Furthermore, in any case, the URL and the credentials absolutely must be runtime properties. Does that make sense?

Reply:
I do not understand your point. in general because as I said, people never change url of a database at runtime. It make no sence for an ops. Maybe to add a cluster but that all.
And advantage of quarkus is that quick reload.

About 1.0 point, I understand that you want to keep same structure but about behaviour, many will change over time. You can not stuck every thing just because it is post 1.0.
A project writter previously before 1.0 will still work with that change. The only impact is that hot change of mongo client is not possible anymore. But in exchange, we reduce the load time....

1.0 do not mean that every thing must be stuck in the marble. Many project change with time. Gnome project for example have even do a full switch of SDK.

Switching back to runtime mean do a fullrework to remove generated class and load all at runtime. I have no time for a such change. So if you still want to block that change, close my PR. Someone else will redo it with runtime mode (slower start).

Reply:
I am sorry but I full disagree here. The URL and credentials for a database are almost always set at runtime, not build time.

Furthermore, we might have to break things post 1.0, but only for a very good reason. And to be honest, this situation doesn't warrant such a break in my view.

I am truly sorry that you think it should be this way and don't like what I am proposing and I completely understand if you don't want to make the change.

Let's have another opinion on this before making any decisions, @gsmet what do you think?

Reply:
Maybe both of you talk about a different *runtime* concept.

@dufoli for Quarkus, runtime means at startup time. These are configuration that are overridable when launching your application via -Dkey=value or via an env file (when will be available). If not overridable at runtime it means that after building your jar file you cannot change the configuration.

I'm agree that database URL must be overridable at runtime.

Reply:
Yeah, @geoand doesn't mean changing the database URL after startup but to be able to provide an adjusted URL at startup and that is non negotiable.

Think provided by an environment variable for instance. This is supported by our config framework.

If set at build time, it's set in stone in the jar/native executable.

Reply:
I think I have a guess where the confusion is coming from and I say it's from the name `BUILD_AND_RUN_TIME_FIXED`. What that means is Quarkus is that you can read the property at runtime from MicroProfile Config, but a user cannot override it using what @loicmathieu mentioned (like `-Dquarkus.mongo.url`).
Does this help explain my point @dufoli ?

Reply:
OK I got the point, now. So in order to build injection with named connection. Best is to have list of named connection on build time but connection string on runtime. I propose to use code to find named mongo client injection and generate producer class from that and keep config at runtime. Wdyt? Can I generate class after index with arc?

Reply:
That sounds about right, but I don't understand the question however. With gizmo you can generate any class and either make it a bean or a regular class depending on the build item you use.
If you set things up properly, Arc can use the generated beans.

Reply:
by the way agroal use fixed runtime too. So it must be changed no ?

Reply:
I don't think that is accurate. Agroal configuration is split into build-time (`AgroalBuildTimeConfig`)  and run-time configuration (`AgroalRuntimeConfig`) the latter of which contains the URL and uses `@ConfigRoot(name = "datasource", phase = ConfigPhase.RUN_TIME)`

Reply:
done